tango_simlib/utilities/validate_device.py

```diff
@@ -0,0 +1,357 @@
+        Tango device name in the domain/family/member format or the
+        FQDN tango://<TANGO_HOST>:<TANGO_PORT>/domain/family/member
+
+    bidirectional: bool
+        Whether to include details on the device that is not in the specification
+
+    Returns
+    -------
+    str
+        The validation result
+    """
+    response = requests.get(url_to_yaml_file, allow_redirects=True)
+    response.raise_for_status()
+    return compare_data(
+        response.text, get_device_specification(tango_device_name), bidirectional
+    )
+
+
+def validate_device_from_path(tango_device_name, path_to_yaml_file, bidirectional):
+    """Retrieves the YAML from the file and checks conformance against the Tango device.
+
+    Parameters
+    ----------
+    path_to_yaml_file : str
+        The path to the specification file
+
+    tango_device_name : str
+        Tango device name in the domain/family/member format or the
+        FQDN tango://<TANGO_HOST>:<TANGO_PORT>/domain/family/member
+
+    bidirectional: bool
+        Whether to include details on the device that is not in the specification
+
+    Returns
+    -------
+    str
+        The validation result
+    """
+    file_path = Path(path_to_yaml_file)
+    assert file_path.is_file(), "{} is not a file".format(file_path)
+    file_data = ""
+    with open(str(file_path), "r") as data_file:
+        file_data = data_file.read()
+    return compare_data(
+        file_data, get_device_specification(tango_device_name), bidirectional
+    )
+
+
+def get_device_specification(tango_device_name):
+    """Translate a device to YAML specification
+
+    Parameters
+    ----------
+    tango_device_name : str
+        Tango device name in the domain/family/member format or the
+        FQDN tango://<TANGO_HOST>:<TANGO_PORT>/domain/family/member
+
+    Returns
+    -------
+    str
+        The device specification in YAML format
+    """
+    parser = TangoToYAML(TangoDeviceParser)
+    return parser.build_yaml_from_device(tango_device_name)
+
+
+def compare_data(specification_yaml, tango_device_yaml, bidirectional):
+    """Compare 2 sets of YAML built from the specification and from the device
+
+    Parameters
+    ----------
+    specification_yaml : str
+        The specification in YAML format
+
+    tango_device_yaml : str
+        The Tango device in YAML format
+
+    bidirectional: bool
+        Whether to include details on the device that is not in the specification
+
+    Returns
+    -------
+    str
+        The validation result
+    """
+    specification_data = yaml.load(specification_yaml, Loader=yaml.FullLoader)
+    tango_device_data = yaml.load(tango_device_yaml, Loader=yaml.FullLoader)
+    if isinstance(specification_data, list):
+        specification_data = specification_data[0]
+    if isinstance(tango_device_data, list):
+        tango_device_data = tango_device_data[0]
+
+    issues = []
+
+    # Class
+    if tango_device_data["class"] != specification_data["class"]:
```

Comment:
As discussed with @SKAJohanVenter,  it would be nice if class checking was also optional.  If key not in spec, then ignore.